kunickiaj/stemcell
# Add support for specifying --cpu-options

We want to try out `--cpu-options` in order to test behavior when disabling multi-threading on various AWS instances: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-specify-cpu-options.html

This requires the low-level AWS SDK v3 API change that @kunickiaj introduces in their PR: https://github.com/airbnb/stemcell/pull/107

This API allows you to specify these options: https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Client.html#run_instances-instance_method
```
    cpu_options: {
      core_count: 1,
      threads_per_core: 1,
    },
 ```
 
 To test this change:
 ```
 gem build stemcell.gemspec
 gem install ./stemcell-0.13.1.gem
 ```
 Then:
 ```
 (AssumedRole=prod-admin) bash-3.2$ stemcell --contexts prod --git-branch production --availability-zone us-east-1e --subnet subnet-66571049 --chef-data-bag-secret-path /Users/melanie_cebula/airlab/repos/chef/cookbooks/airbnb-base/files/default/Chef-Testbox.production.sec.json  --count 1 --instance-type t3.medium --chef-role testbox-amzn --cpu-options="core_count=1,threads_per_core=1"
 ```
 
 Verify:
 ```
 (AssumedRole=prod-admin) bash-3.2$ aws ec2 describe-instances --instance-ids i-0faf7098fe6213d9d | grep Core
                        "CoreCount": 1,
                        "ThreadsPerCore": 1
```

p.s. added ruby `2.6.8` to run locally on my Apple M1 (arm64 support)
@kunickiaj
